qucumber/utils/training_statistics.py

```diff
@@ -107,6 +107,57 @@ def rotate_psi(nn_state, basis, space, unitaries, psi=None):
         psi_r[:, x] = Upsi
     return psi_r
 
+def NLL(nn_state, samples, space, train_bases=None, **kwargs):
+    r"""A function for calculating the negative log-likelihood.
+
+    :param nn_state: The neural network state (i.e. complex wavefunction or
+                     positive wavefunction).
+    :type nn_state: WaveFunction
+    :param samples: Samples to compute the NLL on.
+    :type samples: torch.Tensor
+    :param space: The hilbert space of the system.
+    :type space: torch.Tensor
+    :param train_bases: An array of bases where measurements were taken.
+    :type train_bases: np.array(dtype=str)
+    :param \**kwargs: Extra keyword arguments that may be passed. Will be ignored.
+
+    :returns: The Negative Log-Likelihood.
+    :rtype: torch.Tensor
+    """
+    psi_r = torch.zeros(
+        2, 1 << nn_state.num_visible, dtype=torch.double, device=nn_state.device
+    )
+    NLL = 0.0
+    unitary_dict = unitaries.create_dict()
+    Z = nn_state.compute_normalization(space)
+    eps = 0.000001
+    samples = torch.tensor(samples)
+    if train_bases is None:
+        for i in range(len(samples)):
+            NLL -= (cplx.norm_sqr(nn_state.psi(samples[i])) + eps).log()
+            NLL += Z.log()
+    else:
+        for i in range(len(samples)):
+            # Check whether the sample was measured the reference basis
+            b_ID = 0
```

Comment:
Not super important at the moment, but can you think of a more descriptive name for this variable? Also using a boolean would make this slightly more readable.